test/jdk/java/foreign/CallGeneratorHelper.java

```diff
@@ -145,6 +145,7 @@ MemoryLayout layout(List<StructFieldType> fields) {
             if (this == STRUCT) {
                 long offset = 0L;
                 List<MemoryLayout> layouts = new ArrayList<>();
+                long align = 0;
```

Comment:
I realize this is copied from upstream, but it's not a reasonable value given the `%` operation below on line 163; it should be 8.

Also, I don't think the padding is correct (another error inherited from upstream). If `padding != 0`, the required padding will be `align - padding` (not `padding`).

Reply:
Talked to @keithc-ca offline, we will verify the proposed PR by Keith at first to ensure it really works on all supported platforms. If everything works fine, we will need to bring this up to OpenJDK via the mailing list to request for changes in this method given this method is totally in OpenJDK test code rather than ours.